language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FieldCompletionItemBuilder.java

```diff
@@ -67,4 +67,21 @@ public static CompletionItem build(RecordFieldSymbol symbol) {
     public static CompletionItem build(ObjectFieldSymbol symbol) {
         return getCompletionItem(symbol);
     }
+
+    /**
+     * Build the constant {@link CompletionItem}.
+     *
+     * @param objectFieldSymbol {@link ObjectFieldSymbol}
+     * @param symbolName  Name of the symbol
+     * @return {@link CompletionItem} generated completion item
+     */
+    public static CompletionItem buildField(ObjectFieldSymbol objectFieldSymbol, String symbolName) {
```

Comment:
Without using a separate method name, use an overloaded version of `build(ObjectFieldSymbol symbol)`. Also, use an overloaded version of `getCompletionItem` as well. So you can reuse the generation logics

Reply:
@nadeeshaan Since the symbol passed is always `self`, the method signature reduces to `build(ObjectFieldSymbol objectFieldSymbol)`. Therefore, is it okay to use a separate method name?
Or would you recommend going for an overloaded version of `build(ObjectFieldSymbol symbol)` with  `build(ObjectFieldSymbol objectFieldSymbol, String symbolName)`, passing `self` as the `symbolName`?

Reply:
I would say to go with `build(ObjectFieldSymbol objectFieldSymbol, boolean withSelfPrefix)` with the additional flag. Then, if the `withSelfPrefix` is false, we can route to `build(ObjectFieldSymbol symbol)` and otherwise we implement the `buildField` logic. WDYT?

Reply:
+1.